Sources_v3/StreamChat/Models/Attachment.swift

```diff
@@ -0,0 +1,310 @@
+    /// An author.
+    public let author: String?
+    /// A description text.
+    public let text: String?
+    /// A type (see `AttachmentType`).
+    public let type: AttachmentType
+    /// Actions from a command (see `Action`, `Command`).
+    public let actions: [AttachmentAction]
+    /// A URL.
+    public let url: URL?
+    /// An image preview URL.
+    public let imageURL: URL?
+    /// A file description (see `AttachmentFile`).
+    public let file: AttachmentFile?
+    /// An extra data for the attachment.
+    public let extraData: ExtraData
+    
+    // MARK: - Hashable
+    
+    var hash: String {
+        [title, author, text, type.rawValue, url?.absoluteString, imageURL?.absoluteString]
+            .compactMap { $0 }
+            .reduce("", +)
+    }
+    
+    public static func == (lhs: _ChatMessageAttachment<ExtraData>, rhs: _ChatMessageAttachment<ExtraData>) -> Bool {
+        lhs.title == rhs.title
+            && lhs.author == rhs.author
+            && lhs.text == rhs.text
+            && lhs.type == rhs.type
+            && lhs.url == rhs.url
+            && lhs.imageURL == rhs.imageURL
+            && lhs.file == rhs.file
+    }
+    
+    public func hash(into hasher: inout Hasher) {
+        hasher.combine(title)
+        hasher.combine(author)
+        hasher.combine(text)
+        hasher.combine(type)
+        hasher.combine(url)
+        hasher.combine(imageURL)
+        hasher.combine(file)
+    }
+}
+
+/// An attachment action, e.g. send, shuffle.
+public struct AttachmentAction: Codable {
+    /// A name.
+    public let name: String
+    /// A value of an action.
+    public let value: String
+    /// A style, e.g. primary button.
+    public let style: ActionStyle
+    /// A type, e.g. button.
+    public let type: ActionType
+    /// A text.
+    public let text: String
+    
+    /// Check if the action is cancel button.
+    public var isCancelled: Bool { value == "cancel" }
+    /// Check if the action is send button.
+    public var isSend: Bool { value == "send" }
+    
+    /// Init an attachment action.
+    /// - Parameters:
+    ///   - name: a name.
+    ///   - value: a value.
+    ///   - style: a style.
+    ///   - type: a type.
+    ///   - text: a text.
+    public init(
+        name: String,
+        value: String,
+        style: ActionStyle,
+        type: ActionType,
+        text: String
+    ) {
+        self.name = name
+        self.value = value
+        self.style = style
+        self.type = type
+        self.text = text
+    }
+    
+    /// An attachment action type, e.g. button.
+    public enum ActionType: String, Codable {
+        case button
+    }
+
+    /// An attachment action style, e.g. primary button.
+    public enum ActionStyle: String, Codable {
+        case `default`
+        case primary
+    }
+}
+
+public enum AttachmentType: RawRepresentable, Codable, Hashable, ExpressibleByStringLiteral {
+    /// An attachment type.
+    case unknown
```

Comment:
probably redundant when we have `custom`?

Reply:
Will clarify this also cause maybe `unknown` is some legit type coming from backend.